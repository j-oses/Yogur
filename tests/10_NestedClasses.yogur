// Una jerarquía de clases un poco más profunda, principalmente para probar los accesos '.' anidados

class Top {
	var spinning : Bool
}

class Safe {
	var totem : Top
	def getSpin() -> real : Bool {
		real = totem.spinning
	}
}

class House {
	var safe : Safe
	def getSpin() -> real : Bool{
		real = safe.getSpin()
	}
}

class World {
	var house : House

	def nature_of_reality() -> reality : Bool {
		var some_reality : Bool = house.safe.totem.spinning
		var another_reality : Bool = house.safe.getSpin()
		var ya_reality : Bool = house.getSpin()
		if(another_reality){
			reality = ya_reality
		} else{
			reality = some_reality
		}
	}
}

var success: Bool
var tis : World
tis.house.safe.totem.spinning = true
success = tis.nature_of_reality()