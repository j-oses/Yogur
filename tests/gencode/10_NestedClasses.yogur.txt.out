Run successfully completed
Final P-machine state:
PC = 70   MP = 0   SP = 1   EP = -1   NP = 100   code size = 71
STORE = 
(0,Bool True)
(1,Bool True)
(2,Int 0)
(3,Bool True)
(4,Int 0)
(5,Int 0)
(6,Int (-1))
(7,Int 69)
(8,Int 1)
(9,Bool True)
(10,Bool True)
(11,Bool True)
(12,Int 3)
(13,Bool True)
(14,Int 0)
(15,Int 3)
(16,Int 18)
(17,Int 44)
(18,Int 1)
(19,Int 13)
(20,Bool True)
(21,Int 0)
(22,Int 13)
(23,Int 25)
(24,Int 19)
(25,Int 1)
(26,Int 20)
(27,Bool True)
(28,Null)
(29,Null)
(30,Null)
(31,Null)
(32,Null)
(33,Null)
(34,Null)
(35,Null)
(36,Null)
(37,Null)
(38,Null)
(39,Null)
(40,Null)
(41,Null)
(42,Null)
(43,Null)
(44,Null)
(45,Null)
(46,Null)
(47,Null)
(48,Null)
(49,Null)
(50,Null)
(51,Null)
(52,Null)
(53,Null)
(54,Null)
(55,Null)
(56,Null)
(57,Null)
(58,Null)
(59,Null)
(60,Null)
(61,Null)
(62,Null)
(63,Null)
(64,Null)
(65,Null)
(66,Null)
(67,Null)
(68,Null)
(69,Null)
(70,Null)
(71,Null)
(72,Null)
(73,Null)
(74,Null)
(75,Null)
(76,Null)
(77,Null)
(78,Null)
(79,Null)
(80,Null)
(81,Null)
(82,Null)
(83,Null)
(84,Null)
(85,Null)
(86,Null)
(87,Null)
(88,Null)
(89,Null)
(90,Null)
(91,Null)
(92,Null)
(93,Null)
(94,Null)
(95,Null)
(96,Null)
(97,Null)
(98,Null)
(99,Null)
